core/src/main/java/org/apache/iceberg/PositionDeletesTable.java

```diff
@@ -132,6 +138,35 @@ private Schema calculateSchema() {
                 Types.StringType.get(),
                 MetadataColumns.FILE_PATH_COLUMN_DOC));
 
+    // Calculate used ids (for de-conflict)
+    Set<Integer> currentlyUsedIds =
+        Collections.unmodifiableSet(TypeUtil.indexById(Types.StructType.of(columns)).keySet());
+    Set<Integer> usedIds =
+        table().schemas().values().stream()
+            .map(currSchema -> TypeUtil.indexById(currSchema.asStruct()).keySet())
+            .reduce(currentlyUsedIds, Sets::union);
+
+    // Calculate ids to reassign
+    Set<Integer> idsToReassign =
```

Comment:
Probably can drop the identical version of this you store as metadataIds on 142

Reply:
Good catch